test/common/upstream/subset_lb_test.cc

```diff
@@ -1896,21 +1896,33 @@ TEST_F(SubsetLoadBalancerTest, EnabledLocalityWeightAwareness) {
           {"tcp://127.0.0.1:84", {{"version", "1.0"}}},
           {"tcp://127.0.0.1:85", {{"version", "1.1"}}},
       },
-      host_set_, {1, 100});
+      host_set_, {1, 99});
 
   lb_ = std::make_shared<SubsetLoadBalancer>(
       lb_type_, priority_set_, nullptr, stats_, stats_store_, runtime_, random_, subset_info_,
       ring_hash_lb_config_, maglev_lb_config_, least_request_lb_config_, common_config_);
 
   TestLoadBalancerContext context({{"version", "1.1"}});
 
-  // Since we respect locality weights, the second locality is selected.
-  EXPECT_CALL(random_, random()).WillOnce(Return(0));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][0], lb_->chooseHost(&context));
+  // Verify we respect locality weights.
+  const size_t iters = WRSQScheduler<int>::accuracy();
+  std::vector<size_t> counts({0, 0});
+  for (size_t i = 0; i < iters; ++i) {
+    EXPECT_CALL(random_, random()).WillRepeatedly(Return(i));
+    auto h = lb_->chooseHost(&context);
+    auto& hosts0 = host_set_.healthy_hosts_per_locality_->get()[0];
+    auto& hosts1 = host_set_.healthy_hosts_per_locality_->get()[1];
+    if (std::count(hosts0.begin(), hosts0.end(), h) == 1) {
+      counts[0]++;
+    } else if (std::count(hosts1.begin(), hosts1.end(), h) == 1) {
+      counts[1]++;
+    }
+  }
```

Comment:
the loop above seems to appear 3 times in this file.  Worth adding a helper method?